src/router/routes/tenantBaseMainRoute.ts

```diff
@@ -0,0 +1,181 @@
+            }
+            const urlParts = path.split('/');
+
+            // /{tenantId}/resourceType
+            if (this.options.tenantUrlPart === undefined) {
+                if (validateTenantBaseUrl(0, 1, httpMethod, path)) {
+                    return `/${urlParts.splice(1).join('/')}`;
+                }
+                throw new InvalidResourceError(
+                    `Malformed based url: ${baseUrl} for HTTP method: ${httpMethod}. Expecting /{tenantId}/resourceType/...`,
+                );
+            } // /<tenantUrlPart>/{tenantId}/resourceType
+            else {
+                if (urlParts[0] === this.options.tenantUrlPart) {
+                    if (validateTenantBaseUrl(1, 2, httpMethod, path)) {
+                        return `/${urlParts.splice(2).join('/')}`;
+                    }
+                }
+                throw new InvalidResourceError(
+                    `Malformed based url: ${baseUrl} for HTTP method: ${httpMethod}. Expecting /${this.options.tenantUrlPart}/{tenantId}/resourceType/...`,
+                );
+            }
+        });
+    }
+
+    // Registers a resource url as part of a tenant url
+    use(resourceUrl: string, childRouter: Router): MainRoute {
+        let route;
+        if (this.options.tenantUrlPart !== undefined) {
+            route = `/:tenantType(${this.options.tenantUrlPart})/:tenantId${resourceUrl}`;
+        } else {
+            route = `/:tenantId${resourceUrl}`;
+        }
+
+        if (this.options.tenantAccessTokenClaim !== undefined && !resourceUrl.includes('/metadata')) {
+            this.mainRouter.use(
+                route,
+                (req: express.Request, res: express.Response, next: express.NextFunction) => {
+                    // Check in the authorized tenants of this user with tenant the user requests
+                    if (req.params.tenantId === 'DEFAULT') {
+                        next();
+                    } else if (
+                        this.options.tenantAccessTokenClaim !== undefined &&
+                        res.locals.userIdentity[this.options.tenantAccessTokenClaim] !== undefined
+                    ) {
+                        const tenants: string[] = res.locals.userIdentity[this.options.tenantAccessTokenClaim] ?? [];
+                        const tenant: string =
+                            this.options.tenantAccessTokenClaimValuePrefix !== undefined
+                                ? this.options.tenantAccessTokenClaimValuePrefix + req.params.tenantId
+                                : req.params.tenantId;
+                        if (tenants.includes(tenant) || req.params.tenantId === 'DEFAULT') {
+                            next();
+                        } else {
+                            throw new UnauthorizedError('Unauthorized');
+                        }
+                    } else {
+                        throw new UnauthorizedError('Unauthorized wrong token claim');
```

Comment:
We need to add an if to allow 'cdh/m2m' to continue independently of the tenantid. The scope to use (in this case 'cdh/m2m') shall be configurable